### ISSUE TITLE
String#concat(java.lang.String) does not work

### ISSUE THREAD

**Issue by ketan** [jruby/jruby#4837]
### Environment

Provide at least:
* `jruby 9.1.13.0 (2.3.3) 2017-09-06 8e1c115 Java HotSpot(TM) 64-Bit Server VM 25.121-b13 on 1.8.0_121-b13 [darwin-x86_64]`
* `Darwin ketanpkr.local 15.6.0 Darwin Kernel Version 15.6.0: Sun Jun  4 21:43:07 PDT 2017; root:xnu-3248.70.3~1/RELEASE_X86_64 x86_64 i386 MacBookPro11,4 Darwin`

### Expected Behavior

* Given the interop between java and jruby, I should be able to `String#concat` a (j)ruby string with a java string. Same thing happens when using `String#<<`

Here is a test case to reproduce:

```
$ bin/jruby -e '"ruby string".concat(java.lang.String.new("java string"))'
TypeError: no implicit conversion of Java::JavaLang::String into String
  concat at org/jruby/RubyString.java:2268
  <main> at -e:1
```

**Comment by headius** [comment]
Huh, surprising we don't define to_str and friends on java.lang.String.

This works:

```ruby
class java::lang::String
  alias to_str to_s
end

"foo".concat("bar".to_java(:string))
```

**Comment by ketan** [comment]
Should this go in to jruby? Happy to submit a patch.

**Comment by headius** [comment]
@ketan This has been fixed for JRuby 9.1.15.0.
